Ghidra/Features/Base/src/main/java/ghidra/app/util/bin/format/omf/OmfFixupRecord.java

```diff
@@ -17,312 +17,164 @@
-			lastLEData = null;
-		}
+	/**
+	 * @param last The Datablock this fixup record is meant for
+	 */
+	public void setDataBlock(OmfData last) {
+		lastData = last;
 	}
 
-	public Subrecord[] getSubrecords() {
-		return subrecs;
+	/**
+	 * @return The datablock this fixup record is meant for
+	 */
+	public OmfData getDataBlock() {
+		return lastData;
 	}
 
-	public static class FixupState {
-		public Language language;
-		OmfFileHeader header;
-		public ThreadSubrecord[] frameThreads = new ThreadSubrecord[4];
-		public ThreadSubrecord[] targetThreads = new ThreadSubrecord[4];
-		public OmfFixupRecord currentFixupRecord;
-		public ArrayList<OmfGroupRecord> groups;
-		public ArrayList<OmfSymbol> externals;
-		public int frameState;			// Frame of item being referred to
-		public long targetState;		// Address of item being referred to
-		public Address locAddress;		// Location of data to be patched
-		public boolean M;				// true for segment-relative, false for self-relative
-		public int locationType;
-
-		public FixupState(OmfFileHeader header, ArrayList<OmfSymbol> externsyms, Language lang) {
-			for (int i = 0; i < 4; ++i) {
-				frameThreads[i] = null;
-				targetThreads[i] = null;
-			}
-			this.header = header;
-			groups = header.getGroups();
-			externals = externsyms;
-			language = lang;
-		}
-
-		public void clear() {
-			targetState = -1;
-			locAddress = null;
-			locationType = -1;
-		}
+	/**
+	 * @return The array of subrecords
+	 */
+	public Subrecord[] getSubrecords() {
+		return subrecs;
 	}
 
 	public static class Subrecord {
-		private boolean isThread;
-
-		public Subrecord(boolean isthread) {
-			isThread = isthread;
-		}
-
-		public boolean isThread() {
-			return isThread;
-		}
-	}
-
-	public static class ThreadSubrecord extends Subrecord {
-		private byte type;
+		private byte first;
+		private byte hiFixup;
+		private byte fixData;
 		private int index;
+		private int frameDatum;
+		private int targetDatum;
+		private int targetDisplacement;
 
-		public ThreadSubrecord() {
-			super(true);
-		}
-
-		public int getMethod() {
-			return (type >> 2) & 7;
+		/**
+		 * Read the next subrecord from the input reader
+		 * 
+		 * @param reader The input file
+		 * @param hasBigFields Is this 16 or 32 bit values
+		 * @return The read subrecord
+		 * @throws IOException
+		 */
+		public static Subrecord readSubrecord(BinaryReader reader, boolean hasBigFields)
+				throws IOException {
+			int method;
+			Subrecord rec = new Subrecord();
+			rec.first = reader.readNextByte();
+			rec.index = -1;
+			if (rec.isThreadSubrecord()) {
+				method = rec.getThreadMethod();
+				if (method < 4)
+					rec.index = OmfRecord.readInt1Or2(reader, hasBigFields);
+				return rec;
+			}
+			rec.targetDisplacement = 0;
+			rec.targetDatum = 0;
+			rec.hiFixup = reader.readNextByte();
+			rec.fixData = reader.readNextByte();
+			method = rec.getFrameMethod();
+			if ((rec.fixData & 0x80) == 0 && method < 3) 	// F=0  (explicit frame method (and datum))
```

Comment:
these if()'s are badly formatted.  The formatter should be enforcing something like "if(....) {\n statement(); \n}", or it might be one of those settings that isn't codified in the xml and you need to hand set it... I'm not sure.

Reply:
I've confirmed our formatter is not adding the curly braces for some reason any more.  I'll get that fixed.

Reply:
It turns out this is a Save Actions or Code Cleanup option, not a formatter option,

Reply:
Those can be imported/exported as well. :)
So my suggestion is that gets added with the other configurations.